### PR TITLE
Add examples to Exception object and group members

### PR DESCRIPTION
**Pull request by janekdb**
- Extend main comment with additional examples
- Group methods from a user perspective
- List exceptions special cased by shouldRethrow
- Include overlooked withTry in opt, either notes.

### CODE REVIEW

**Comment by janekdb** [comment]
scaladoc-exception-examples-groups-v1

File: src/library/scala/util/control/Exception.scala [hunk 1]
```diff
@@ -178,42 +290,56 @@ object Exception {
 
   /** Creates a `Catch` object which will catch any of the supplied exceptions.
    *  Unlike "catching" which filters out those in shouldRethrow, this one will
-   *  catch whatever you ask of it: `ControlThrowable`, `InterruptedException`,
-   *  `OutOfMemoryError`, you name it.
+   *  catch whatever you ask of it including `ControlThrowable` or `InterruptedException`.
```
Comment (janekdb) [inline src/library/scala/util/control/Exception.scala]: Should be $protectedExceptions

**Comment by lrytz** [comment]
meh, jenkins doesn't want to pick this one up.. maybe re-submit as a new PR?

**Comment by janekdb** [comment]
Replaced by #5111

**Pull request closed without merging**